service/awsconfig/v5/resource/cloudformation/adapter/adapter.go

```diff
@@ -66,7 +67,14 @@ type Config struct {
 }
 
 func NewGuest(cfg Config) (Adapter, error) {
-	a := Adapter{}
+	a := Adapter{
+		Outputs: &outputsAdapter{},
+	}
+
+	err := a.Outputs.Adapt(cfg)
+	if err != nil {
+		return Adapter{}, microerror.Mask(err)
+	}
```

Comment:
The adapter has to be called "manually" because of the scoping. There is no option to generalize this with a loop like below.

Reply:
I don't understand this, given its signature. `a.Outputs.Adapt` is an `hydrater`, why can't you include it in the loop below?

Reply:
Good catch. I realized we can do this just how it was and add it to the loop again. Not sure what I thought when I wrote this or what was holding me back here.